platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/location/LocationLayerController.java

```diff
@@ -262,6 +270,9 @@ boolean onMapClick(@NonNull LatLng point) {
     new MapLibreAnimator.AnimationsValueChangeListener<LatLng>() {
       @Override
       public void onNewAnimationValue(LatLng value) {
+        if (puckPositionUpdateListener != null) {
+          puckPositionUpdateListener.onPuckPositionChanged(value);
```

Comment:
@ahmedre is there a reason you trigger the listener before the update to the renderer happens? Maybe changing this to after could fix some of the delays that happen in the after video in the issue https://linear.app/hudhudapp/issue/HHAND-1088/path-eating-and-puck-animation